src/batch/src/task/context.rs

```diff
@@ -68,52 +68,20 @@ pub trait BatchTaskContext: Clone + Send + Sync + 'static {
     fn mem_usage(&self) -> usize;
 
     fn create_executor_mem_context(&self, executor_id: &str) -> Option<MemoryContextRef>;
-    fn get_stop_flag(&self) -> Arc<StopFlag>;
 
-    fn get_stop_flag_ref(&self) -> &StopFlag;
-}
-
-/// We already have `shut_down` sender to stop execution.
-/// `StopFlag` used in executor to let it stop execute more timely when it can't receive message
-/// timely.
-#[derive(Clone)]
-pub struct StopFlag {
-    stop_flag: Arc<AtomicBool>,
-    stop_reason: Arc<RwLock<Option<String>>>,
-}
-
-impl StopFlag {
-    pub fn new() -> Self {
-        Self {
-            stop_flag: Arc::new(AtomicBool::new(false)),
-            stop_reason: Arc::new(RwLock::new(None)),
-        }
-    }
+    fn send_shutdown(
+        &self,
+        msg: ShutdownMsg,
+    ) -> std::result::Result<(), error::SendError<ShutdownMsg>>;
```

Comment:
`BatchTaskContext` is usually used by `Executor`, so we should not put this method here.